pallets/liquid-staking/src/lib.rs

```diff
@@ -1732,6 +1765,75 @@ pub mod pallet {
             Ok(())
         }
 
+        #[require_transactional]
+        fn do_fast_match_unstake(unstaker: &T::AccountId) -> DispatchResult {
+            FastUnstakeRequests::<T>::try_mutate_exists(unstaker, |b| -> DispatchResult {
+                if let Some(request_amount) = b.take() {
+                    let available_amount = Self::matching_pool().total_stake_amount.free()?;
+                    let actual_amount = available_amount.min(request_amount);
```

Comment:
so we are allowing partially matched right

Reply:
Yes